src/providers/ipa/ipa_access.c

```diff
@@ -470,83 +471,63 @@ static errno_t ipa_fetch_hbac_recv(struct tevent_req *req)
     return EOK;
 }
 
-static errno_t ipa_save_hbac(struct sss_domain_info *domain,
+static errno_t ipa_save_hbac(TALLOC_CTX *mem_ctx,
+                             struct sss_domain_info *domain,
                              struct ipa_fetch_hbac_state *state)
```

Comment:
`ipa_save_hbac` as of this commot leaks memory because it allocates everything on mem_ctx, but then never frees the temporary data. But I'm not sure if it's worth fixing because the resulting code is different?